### PR TITLE
Handle undefined values

### PR DESCRIPTION
**Pull request by vesln**

### CODE REVIEW

File: lib/parser.js [hunk 1]
```diff
@@ -128,14 +128,31 @@ Parser.prototype._parsePair = function(str) {
...
 /**
+ * Parse undefined value.
+ *
+ * @param {Object} node
+ * @returns {Object}
+ * @private
+ */
+
+Parser.prototype._parseUndefined = function(node) {
+  if (node.val === void 0) {
```
Comment (logicalparadox) [inline lib/parser.js]: Been awhile since i have seen this.

**Comment by logicalparadox** [comment]
:+1:

**Comment by wilmoore** [comment]
Will an undefined line also still handle a comment on that line?

```
FOO=    // this is required!
```

**Comment by wilmoore** [comment]
:+1:

**Comment by vesln** [comment]
@wilmoore adding a test for it

File: lib/parser.js (hunk 1 above)
Comment (n3bulous) [inline lib/parser.js]: I've never seen that...  I want to call BS :)

Comment (wilmoore) [inline lib/parser.js]: LOL @n3bulous 

It's just another way to say `undefined`. It used to be promoted as a way to be safe against those that might overwrite `undefined` (yes, possible in non-strict mode); however, not something people worry too much about these days. At the same time, I have no problem with it when used. Not to mention, it's shorter than `undefined`. Not that it matters, but I guess that's a benefit :smile:

File: test/fixtures/undefined [hunk 2]
```diff
@@ -0,0 +1,2 @@
+FOO=
+BAR= # This is a comment
```
Comment (wilmoore) [inline test/fixtures/undefined]: :+1:

**Pull request merged**